### ISSUE TITLE
Tokyo map is too polluted with subway entrances

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#5837]
@pastk can we show entrances at one zoom later?

![image](https://github.com/organicmaps/organicmaps/assets/170263/f47fc8da-8c15-4ca6-a30d-b8f7ec7b35ab)

**Comment by map-per** [comment]
Looks like the starting zoom level for subays is not the same in all citys:
- Munich at z13 (looks good)
- Vienna at z12 (looks cluttered)
- Tokio at z12 (looks cluttered)

**Comment by pastk** [comment]
> Looks like the starting zoom level for subays is not the same in all cities:
> 

Yeap in some cities z12 looks good, in some we display subways at z11 even.
Basically it needs adjustment on per-city level..

If anyone wants to submit a PR - you're very welcome!
Here is a similar PR for Barcelona https://github.com/organicmaps/organicmaps/pull/4693

**Comment by bhavuk2002** [comment]
can you assign me this. i m ready to work. just guide me lit bit, rest i will try my best to figure out, im familiar with JS and Nodejs.

**Comment by RedAuburn** [comment]
> can you assign me this. i m ready to work. just guide me lit bit, rest i will try my best to figure out, im familiar with JS and Nodejs.

We don't assign issues to new contributors, feel free to just submit a PR :)
(JS/NodeJS won't be any use for this I'm afraid, it's just adjusting a stylesheet)

**Comment by bhavuk2002** [comment]
sure why not i will give it a try also say #4693 for reference : ) 
ik this is not right place to ask this but in which language should i contribute? and how to find some initial projects? everything looks complicated.

**Comment by map-per** [comment]
Yes, the project is quite big. 
- I'd start with setting up the development environment as described in [INSTALL.md](https://github.com/organicmaps/organicmaps/blob/ea37cc36565a30695e7a045e83abfb1886b8b538/docs/INSTALL.md).
- Then you can follow [STYLES.md](https://github.com/organicmaps/organicmaps/blob/ea37cc36565a30695e7a045e83abfb1886b8b538/docs/STYLES.md) to replicate the changes made in https://github.com/organicmaps/organicmaps/pull/4693/commits/739948f3b5735a661eb57748978250c9807a5747  for Tokyo

(So the language is customized mapcss)

**Comment by bhavuk2002** [comment]
Okay thanks i will let you know once done. Can you suggest some other project this one seems too complicated?

**Comment by charlie2clarke** [comment]
Currently, the Tokyo subway entrances are visible at z12-13. 

Here is what it looks like at z13:
<img src="https://github.com/organicmaps/organicmaps/assets/58811769/f7681286-5fd4-402e-be36-a2c3e3d893d0" alt="z14" width="50%" height="50%"/>

And here is what it looks like at z14:
<img src="https://github.com/organicmaps/organicmaps/assets/58811769/390d2871-1d60-4bfb-b1bb-1068af844138" alt="z14" width="50%" height="50%"/>

IMO, I think it looks better when visible at z14 - any thoughts?

**Comment by biodranik** [comment]
Maybe make smaller/less visible/more transparent icons on earlier zoom levels? Removing is also ok. It's more important to see district names at that zoom.

**Comment by pastk** [comment]
@charlie2clarke hmm on your screenshots zoom level is the same (z13), but on the second Tokyo-specific subway icons are just replaced with generic subway icons...

**Comment by pastk** [comment]
And actually it would be good to display regular train stations in Tokyo one or even two zooms later also,
e.g. now at z12:
![image](https://github.com/organicmaps/organicmaps/assets/18434508/54b95d23-ec7a-4d20-9c68-ea8f53007332)

**Comment by charlie2clarke** [comment]
> And actually it would be good to display regular train stations in Tokyo one or even two zooms later also, e.g. now at z12: ![image](https://user-images.githubusercontent.com/18434508/274282623-54b95d23-ec7a-4d20-9c68-ea8f53007332.png)

^ I was thinking the same. 

And yes I can see the fault with the previous screenshot - my understanding of the styling process isn't the best, sorry. Is it right that the subway-tokyo icon is overwritten by the subway-s icon if the Subways.mapcss doesn't include rules to show the icon on z13? For example, changing tokyo's styling to below:
```css
/* Tokyo Subway Station */
node|z14[railway=station][transport=subway][city=tokyo] {icon-image: subway-tokyo-s.svg;icon-min-distance: 1;}
node|z15[railway=station][transport=subway][city=tokyo] {icon-image: subway-tokyo-s.svg;}
node|z16-17[railway=station][transport=subway][city=tokyo] {icon-image: subway-tokyo-m.svg;}
node|z18-[railway=station][transport=subway][city=tokyo] {icon-image: subway-tokyo-m.svg;}
node|z16-[railway=subway_entrance][city=tokyo] {icon-image: subway-tokyo-s.svg;}
```
Means it is overwritten by this (in Icons.mapcss):
```css
/* 3.2 Subway Station */

node|z13[railway=station][transport=subway]
{icon-image: subway-s.svg;icon-min-distance: 1;}
```
If so, how can I stop this being overwritten just for the case of Tokyo?

**Comment by pastk** [comment]
At the top of Tokyo's declarations add 
```
node|z13[railway=station][transport=subway][city=tokyo] { icon-image: none; }
```
It'll override and effectively disable the declaration from Icons.mapcss.
(Subways.mapcss is processed after Icons.mapcss)

**Comment by charlie2clarke** [comment]
> Here is what it looks like at z13: <img alt="z13" width="50%" height="50%" src="https://user-images.githubusercontent.com/58811769/274244044-f7681286-5fd4-402e-be36-a2c3e3d893d0.png">

Ah sorry yes, that is what the previous one (above) is showing, but I still think it looks a bit too polluted with them. WDYT?

**Comment by pastk** [comment]
Yeap it looks cluttered.

**Comment by charlie2clarke** [comment]
My apologies, I must've not regenerated the drules previously... 

Making Tokyo subway stations only visible at z13 looks like this:
<img src="https://github.com/organicmaps/organicmaps/assets/58811769/a50cca52-b1e2-4caa-8f9f-1b28a80418c6" alt="z13" width="50%" height="50%" />

Then at z14:
<img src="https://github.com/organicmaps/organicmaps/assets/58811769/6e3d8cd2-a63d-4f10-b765-7294e7d0d083" alt="z14" width="50%" height="50%" />

Should I make a PR?

**Comment by pastk** [comment]
Yeap please submit a PR!

**Comment by charlie2clarke** [comment]
PR submitted! I promise I will try and make my contributions less hands-on. Thanks for your help guys

**Comment by pastk** [comment]
Fixed by #6291
